cli/tests/default/e2e/test_test.py

```diff
@@ -27,21 +42,44 @@ def test_cli_test_basic(run_semgrep_in_tmp: RunSemgrep, snapshot):
     )
 
 
+# It should accept a directory as a target.
+# TODO: this actually does not really test the ability to iterate over
+# a directory (ability heavily used in semgrep-rules/scripts/run-test.sh)
 @pytest.mark.kinda_slow
+def test_cli_test_directory(run_semgrep_in_tmp: RunSemgrep, snapshot):
+    results, _ = run_semgrep_in_tmp(
+        "rules/cli_test/directory/",
+        options=["--test"],
+        target_name="cli_test/directory/",
+        output_format=OutputFormat.JSON,
+    )
+
+    snapshot.assert_match(
+        results,
+        "results.json",
+    )
+
+
+# It should output an "error" field with the right error message (timeout)
+# in the JSON output.
+# TODO: adding "--timeout", "1", does not seem to speedup things
+@pytest.mark.slow
 @pytest.mark.osemfail
 def test_timeout(run_semgrep_in_tmp: RunSemgrep, snapshot):
     results, _ = run_semgrep_in_tmp(
-        "rules/cli_test/error/",
+        "rules/cli_test/rule_that_timeout.yaml",
         options=["--test"],
-        target_name="cli_test/error/",
+        target_name="cli_test/long.py",
         output_format=OutputFormat.JSON,
+        assert_exit_code=1,
     )
     snapshot.assert_match(
         results,
         "results.json",
     )
 
 
+# ??
```

Comment:
I think this is for testing that yaml files can also be targets. (https://github.com/semgrep/semgrep/commit/e2f48700d2a785154106b746fbc8c0167cd12c22)

Reply:
Got it.